dpe/src/commands/certify_key.rs

```diff
@@ -158,91 +161,101 @@ impl CommandExecution for CertifyKeyCmd {
-                    &priv_key,
-                    &pub_key,
-                )?;
-
-                let mut csr_buffer = [0u8; MAX_CERT_SIZE];
-                let mut csr_writer = CertWriter::new(&mut csr_buffer, true);
-                bytes_written = csr_writer
-                    .encode_csr(&cert_req_info_buffer[..bytes_written], &cert_req_info_sig)?;
-                if bytes_written > MAX_CERT_SIZE {
-                    return Err(DpeErrorCode::InternalError);
-                }
+                    let cert_req_info_digest = env.crypto.hash(
+                        DPE_PROFILE.alg_len(),
+                        &cert_req_info_buffer[..bytes_written],
+                    )?;
+                    // The PKCS#10 CSR is self-signed so the private key signs it instead of the alias key.
+                    let cert_req_info_sig = env.crypto.ecdsa_sign_with_derived(
+                        DPE_PROFILE.alg_len(),
+                        &cert_req_info_digest,
+                        &priv_key,
+                        &pub_key,
+                    )?;
+
+                    let mut csr_buffer = [0u8; MAX_CERT_SIZE];
+                    let mut csr_writer = CertWriter::new(&mut csr_buffer, true);
+                    bytes_written = csr_writer
+                        .encode_csr(&cert_req_info_buffer[..bytes_written], &cert_req_info_sig)?;
+                    if bytes_written > MAX_CERT_SIZE {
+                        return Err(DpeErrorCode::InternalError);
+                    }
 
-                let csr_digest = env
-                    .crypto
-                    .hash(DPE_PROFILE.alg_len(), &csr_buffer[..bytes_written])?;
-                let csr_sig = env
-                    .crypto
-                    .ecdsa_sign_with_alias(DPE_PROFILE.alg_len(), &csr_digest)?;
-                let sid = env.platform.get_signer_identifier()?;
-
-                let mut cms_writer = CertWriter::new(&mut cert, true);
-                bytes_written =
-                    cms_writer.encode_cms(&csr_buffer[..bytes_written], &csr_sig, &sid)?;
-                u32::try_from(bytes_written).map_err(|_| DpeErrorCode::InternalError)?
+                    let csr_digest = env
+                        .crypto
+                        .hash(DPE_PROFILE.alg_len(), &csr_buffer[..bytes_written])?;
+                    let csr_sig = env
+                        .crypto
+                        .ecdsa_sign_with_alias(DPE_PROFILE.alg_len(), &csr_digest)?;
+                    let sid = env.platform.get_signer_identifier()?;
+
+                    let mut cms_writer = CertWriter::new(&mut cert, true);
+                    bytes_written =
+                        cms_writer.encode_cms(&csr_buffer[..bytes_written], &csr_sig, &sid)?;
+                    u32::try_from(bytes_written).map_err(|_| DpeErrorCode::InternalError)?
+                }
+                #[cfg(feature = "disable_csr")]
+                Err(DpeErrorCode::InternalError)?
```

Comment:
Done. My initial reasoning was that this code should not be reachable since we have guards earlier in the function to prevent this so I was considering it an internal error. But I think not supported is fine.